pkg/model/api/config/authentication.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"time"
+
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	DefaultAccessTokenTTL  = time.Minute * 15
+	DefaultRefreshTokenTTL = time.Hour * 24
+)
+
+type OIDCProvider struct {
+	SlugID       string            `json:"slug_id" yaml:"slug_id"`
+	Name         string            `json:"name" yaml:"name"`
+	ClientID     config.Secret     `json:"client_id" yaml:"client_id"`
+	ClientSecret config.Secret     `json:"client_secret" yaml:"client_secret"`
+	RedirectURI  string            `json:"redirect_uri" yaml:"redirect_uri"`
+	Scopes       []string          `json:"scopes" yaml:"scopes"`
+	Issuer       config.Secret     `json:"issuer" yaml:"issuer"`
+	URLParams    map[string]string `json:"url_params" yaml:"url_params"`
+}
+
+type OAuthProvider struct {
+	SlugID       string        `json:"slug_id" yaml:"slug_id"`
+	Name         string        `json:"name" yaml:"name"`
+	ClientID     config.Secret `json:"client_id" yaml:"client_id"`
+	ClientSecret config.Secret `json:"client_secret" yaml:"client_secret"`
+	RedirectURI  string        `json:"redirect_uri" yaml:"redirect_uri"`
+	Scopes       []string      `json:"scopes" yaml:"scopes"`
+	AuthURL      string        `json:"auth_url" yaml:"auth_url"`
+	TokenURL     string        `json:"token_url" yaml:"token_url"`
+	LogoutURL    string        `json:"logout_url" yaml:"logout_url"`
+	UserInfosURL string        `json:"user_infos_url" yaml:"user_infos_url"`
+}
+
+type AuthProviders struct {
```

Comment:
based on what I saw in the code, it seems `slug_id` must be unique for the same kind of provider.

Would you mind to implement the method `Verify` for the `AuthProviders` to adresse that please ?

Reply:
Ok I will

Reply:
Done and tested :)